<launch>
  <!-- Load the Optitrack launch file -->
  <include file="$(find vrpn_client_ros)/launch/sample.launch">
      <arg name="server" value="192.168.1.181" />
  </include>  

  <!-- Load the offline IPP node -->
  <node pkg="ros_sgp_ipp" name="offlineIPP" type="offlineIPP.py" output="screen" />
  
  <group ns = "tb3_0">
    <!-- Load the turtlebot3_robot launch file -->
    <include file="$(find turtlebot3_bringup)/launch/turtlebot3_robot.launch" />

    <!-- Load the data publisher node -->
    <node pkg="ros_sgp_ipp" name="data_publisher" type="data_publisher.py" output="screen" />

    <!-- Load the trajectory planner and online IPP nodes -->
    <node pkg="ros_sgp_ipp" name="trajectory_planner" type="trajectory_planner.py" output="screen" />

    <!-- Load the online IPP node -->
    <node pkg="ros_sgp_ipp" name="onlineIPP" type="onlineIPP.py" output="screen" />
  </group>
</launch>